pydicom/pixel_data_handlers/numpy_handler.py

```diff
@@ -134,8 +127,34 @@ def get_pixeldata(dicom_dataset):
             "Amount of pixel data %d does not "
             "match the expected data %d" %
             (length_of_pixel_array, padded_length))
-    if expected_length != padded_length:
+
+    # the checks above confirmed the amount of data in PixelData is what
+    #  we expect, now we can unpack it
+    if dicom_dataset.BitsAllocated == 1:
+        # if single bits are used for binary representation, a uint8 array
+        # has to be converted to a binary-valued array (that is 8 times bigger)
+        bit = 0
+        pixel_array = \
+            numpy.ndarray(shape=(length_of_pixel_array*8), dtype='uint8')
+        # bit-packed pixels are packed from the right; i.e., the first pixel
+        #  in the image frame corresponds to the first from the right bit of
+        #  the first byte of the packed PixelData!
+        #  See the following for details:
+        #  * DICOM 3.5 Sect 8.1.1 (explanation of bit ordering)
+        #  * DICOM Annex D (examples of encoding)
+        for byte in pixel_bytearray:
+            for bit in range(bit, bit+8):
+                pixel_array[bit] = byte & 1
+                byte >>= 1
+            bit += 1
```

Comment:
This fails in Python 2 because `byte` is a `str`. You may need to convert it to `int` using `ord` or something.

Reply:
Done!